src/main/java/com/pixonic/ctop/Main.java

```diff
@@ -1,45 +1,105 @@
+        String keySpace = properties.getProperty(Constants.CASSDB_KEYSPACE).trim();
+
+        int interval = Integer.parseInt(properties.getProperty(Constants.CONFIG_INTERVAL_SEC, "10"));
+
+        String jmxUsername = properties.getProperty(Constants.CONFIG_JMX_USERNAME).trim();
+        String jmxPassword = properties.getProperty(Constants.CONFIG_JMX_PASSWORD).trim();
 
-        String hostAndPort = args[0];
-        String keySpace = args[1];
-        int interval = 10;
-        if (args.length > 2) {
-            interval = Integer.parseInt(args[2]);
+        boolean hasCreds = false;
+        String[] creds = null;
+        if (jmxUsername != null && jmxPassword != null) {
+            creds = new String[]{jmxUsername, jmxPassword};
+            hasCreds = true;
         }
 
         System.out.println("Connecting to " + hostAndPort + "...");
 
+        System.out.println("======= Properties Loaded =======");
+        System.out.println(properties.toString());
+
+        MetricsType metricsType = MetricsType.valueOf(properties.getProperty("metrics.type", "NONE"));
+
+        MetricsCollector metricsCollector = MetricUtils.registerMetricsReporter(metricsType, properties);
+
         final JMXServiceURL target = new JMXServiceURL("service:jmx:rmi:///jndi/rmi://" + hostAndPort + "/jmxrmi");
-        final JMXConnector connector = JMXConnectorFactory.connect(target);
+
+        HashMap<String, Object> environment = new HashMap<>();
+        environment.put(JMXConnector.CREDENTIALS, creds);
+
+        final JMXConnector connector = JMXConnectorFactory.connect(target, (hasCreds) ? environment : null);
         final MBeanServerConnection remote = connector.getMBeanServerConnection();
 
         ObjectName storageMBean = new ObjectName("org.apache.cassandra.db:type=StorageService");
         String releaseVersion = (String) remote.getAttribute(storageMBean, "ReleaseVersion");
         int majorVersion = Integer.valueOf(releaseVersion.substring(0, releaseVersion.indexOf('.')));
         System.out.println("Cassandra version is " + releaseVersion);
 
-        Metrics metrics = MetricsFactory.getMetrics(majorVersion, interval, remote, keySpace);
+        TargetType targetType = TargetType.valueOf(properties.getProperty(Constants.CONFIG_TARGET_TYPE, "ALL"));
+
+        Metrics metrics = MetricsFactory.getMetrics(majorVersion, interval, remote, keySpace, targetType, metricsType, metricsCollector);
         System.out.println("Connected. Gathering data...");
         metrics.printMetrics();
 
-        Runtime.getRuntime().addShutdownHook(new Thread(metrics::shutdown));
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> metrics.shutdown()));
     }
 
+
+    public static void main(String[] args) throws Exception {
+
+        Properties properties = new Properties();
+
+        //customer file passed so load it
+        if (args.length > 0) {
+            String argument = args[0].trim();
+            if (argument.equals("-h")) {
```

Comment:
I don't like idea that we have to provide config.properties in order to have console metrics.
We should keep previous implementation of ctop for console.

You can implement 2 options:
1. Console option: use program arguments (like in initial implementation of ctop) or config.properties
2. Graphite option: use only config.properties